src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -480,4 +490,64 @@ private String getInvoiceLineNumberEndpoint(String id) {
 
   }
 
+  /**
+   * Updates the invoice's poNumbers field, following an invoice line creation, update or removal.
+   * @param invoice - the invoice of the modified invoice line
+   * @param invoiceLine - the modified invoice line
+   * @param invoiceLineFromStorage - the old version of the invoice line
+   * @param requestContext - used to start new requests
+   */
+  private CompletableFuture<Void> updateInvoicePoNumbers(Invoice invoice, InvoiceLine invoiceLine,
+      InvoiceLine invoiceLineFromStorage, RequestContext requestContext) {
+
+    if (invoiceLineFromStorage == null && invoiceLine.getPoLineId() == null)
+      return CompletableFuture.completedFuture(null);
+    if (invoiceLine.getPoLineId() == null && invoiceLineFromStorage != null && invoiceLineFromStorage.getPoLineId() == null)
+      return CompletableFuture.completedFuture(null);
+    String poLineId = (invoiceLineFromStorage == null || invoiceLine.getPoLineId() != null) ? invoiceLine.getPoLineId() :
+      invoiceLineFromStorage.getPoLineId();
+    return orderService.getPoLine(poLineId, requestContext)
+      .thenCompose(poLine -> orderService.getOrder(poLine.getPurchaseOrderId(), requestContext))
+      .thenCompose(order -> {
+        List<String> invoicePoNumbers = invoice.getPoNumbers();
+        String orderPoNumber = order.getPoNumber();
+        if (orderPoNumber == null)
+          return CompletableFuture.completedFuture(null);
+        if (invoiceLineFromStorage != null && invoiceLineFromStorage.getPoLineId() != null && invoiceLine.getPoLineId() == null) {
+          // remove the po number if needed
+          if (invoicePoNumbers == null || !invoicePoNumbers.contains(orderPoNumber))
```

Comment:
According to ObjectMapper configuration, array will never be `null`, but it can be empty or not. It seems checking condition `invoicePoNumbers == null` is redundant here

Reply:
OK. Where is this configured or documented ?

Reply:
I was wrong. Not the ObjectMapper, but jsonschema2pojo genarator initializes lists by default 
https://joelittlejohn.github.io/jsonschema2pojo/site/1.1.1/generate-mojo.html#initializeCollections

Reply:
I guess what I had trouble to understand was that Vert.x JSON conversion to JsonObject is not setting any map item for something that is explicitely set to null (as opposed to setting it with a null value). So poNumbers is not defined in the JsonObject map, and the mapTo is not changing the field, and it keeps its initial value. I didn't find a documentation about that behavior.

Reply:
I think I finally understand it. Vert.x is using Jackson internally to deserialize JSON. Jackson is basing its behavior (ignoring null values) on the `@JsonInclude(JsonInclude.Include.NON_NULL)` annotation in the generated code. This code is created by default by jsonschema2pojo, which is used by RMB to generate code from the schema. Hopefully this default behavior will not change in the future.